MaintenanceDashboard.Client/ViewModels/RegisterToolViewModel.cs

```diff
@@ -100,22 +94,21 @@ public void GetRegisterToolList()
                 RegisterTools.Add(item);
         }
 
-        private void SaveRegisterTool()
+        private void TrySaveRegisterTool()
```

Comment:
W dwóch funkcjach dopisałem `Try`. Pytanie, dlaczego?

Pisząc `SaveRegisterTool` ty OBIECUJESZ użytkownikowi, że zapiszesz narzędzie rejestracji (?). Zaś w ciele metody nie jest to takie oczywiste, czyli w nazwie metody mówisz, że `Zapisuje` z kolei w kodzie `Zapisuje, ale nie zawsze`. Stąd dodałem słówko `Try...` które mówi mi, że ta metoda SPRÓBUJE ZAPISAĆ daną wartość do bazy i nie obiecuje ci, że to zrobi.

Z takich metod dobrze by było zwrócić `bool` czy operacja się udała czy nie. Zwróc uwagę na systemową metodę `bool i = int.TryParse(string number, out int number)`. Ta metoda nie obiecuje ci, że sparsuje stringa do inta. Ona może spróbować to zrobić i jeśli się uda, zwróci ci `true`. Jeśli nie, `false`

Reply:
Jeśli nie czytałeś, to polecam:

https://tantis.pl/czysty-kod-podrecznik-dobrego-programisty-p459562.html?utm_source=advertisement&utm_medium=google_search&gclid=Cj0KCQjwp4j6BRCRARIsAGq4yMGQnH4KSP3P2lfAbhtTfI0csKSfbZ2-wYSezFd0N9W2gYujIR_sY_AaAvl2EALw_wcB

Nie musisz czytać od dechy do dechy ale przeleć ją. Wiele rzeczy wyjaśnia, jak pisać dobry kod. Elementarz programistów